contributed_definitions/nyaml/NXafm.yaml

```diff
@@ -0,0 +1,269 @@
+category: application
+doc: |
+  An application definition to describe Atomic Force Microscopy (AFM).
+type: group
+NXafm(NXspm):
+  (NXentry):
+    definition: 
+      doc: |
+        Name of the definition that is used for the application.
+      enumeration:
+        - NXafm
+    scan_mode:
+      doc: |
+        The mode of the scan.
+      enumeration: ["tapping mode", "contact mode", "dynamic mode"]
```

Comment:
'non-contact mode' is missing, nowadays less often used, but still matters.
'dynamic mode' do you mean force-modulation?
'quantitative-imaging (QI mode)' depends on the instrument manufacturer, it is the most dynamic generating force curves for every pixel (surface point)

Reply:
`dynamic mode` removed and `non-contact` mode added.

I meant the `non-contact mode` for `dynamic mode`, as `tapping mode` is also called `intermittent contact` mode.  Sometimes, the force modulation method also implements contact mode.